antarest/study/storage/variantstudy/model/command/create_binding_constraint.py

```diff
@@ -206,17 +216,81 @@ def get_corresponding_matrices(
         raise TypeError(repr(v))
 
     def validates_and_fills_matrices(
-        self, *, specific_matrices: t.Optional[t.List[str]], version: int, create: bool
+        self,
+        *,
+        time_step: BindingConstraintFrequency,
+        specific_matrices: t.Optional[t.List[str]],
+        version: int,
+        create: bool,
     ) -> None:
         if version < 870:
-            self.values = self.get_corresponding_matrices(self.values, version, create)
+            self.values = self.get_corresponding_matrices(self.values, time_step, version, create)
         elif specific_matrices:
             for matrix in specific_matrices:
-                setattr(self, matrix, self.get_corresponding_matrices(getattr(self, matrix), version, create))
+                setattr(
+                    self, matrix, self.get_corresponding_matrices(getattr(self, matrix), time_step, version, create)
+                )
         else:
-            self.less_term_matrix = self.get_corresponding_matrices(self.less_term_matrix, version, create)
-            self.greater_term_matrix = self.get_corresponding_matrices(self.greater_term_matrix, version, create)
-            self.equal_term_matrix = self.get_corresponding_matrices(self.equal_term_matrix, version, create)
+            self.less_term_matrix = self.get_corresponding_matrices(self.less_term_matrix, time_step, version, create)
+            self.greater_term_matrix = self.get_corresponding_matrices(
+                self.greater_term_matrix, time_step, version, create
+            )
+            self.equal_term_matrix = self.get_corresponding_matrices(self.equal_term_matrix, time_step, version, create)
+
+    def apply_binding_constraint(
```

Comment:
Some parts of this code could proably be merged into AbstractCommand as now both classes are optional